testdata/cheats/loadAllocs.t.sol

```diff
@@ -6,15 +6,43 @@ import "./Vm.sol";
 
 contract LoadAllocsTest is DSTest {
     Vm constant vm = Vm(HEVM_ADDRESS);
-    string allocsPath;
     address constant ALLOCD = address(0x420);
     address constant ALLOCD_B = address(0x421);
 
+    uint256 snapshotId;
+    string allocsPath;
+
     function setUp() public {
         allocsPath = string.concat(vm.projectRoot(), "/fixtures/Json/test_allocs.json");
+
+        // Snapshot the state; We'll restore it in each test that loads allocs inline.
+        snapshotId = vm.snapshot();
+
+        // Load the allocs file.
+        vm.loadAllocs(allocsPath);
+    }
+
+    /// @dev Checks that the `loadAllocs` cheatcode persists account info if called in `setUp`
+    function testLoadAllocsStaticSetup() public {
+        // Balance should be `0xabcd`
+        assertEq(ALLOCD.balance, 0xabcd);
+
+        // Code should be a simple store / return, returning `0x42`
+        (bool success, bytes memory rd) = ALLOCD.staticcall("");
+        assertTrue(success);
+        uint256 ret = abi.decode(rd, (uint256));
+        assertEq(ret, 0x42);
+
+        // Storage should have been set in slot 0x1, equal to `0xbeef`
+        assertEq(uint256(vm.load(ALLOCD, bytes32(uint256(0x10 << 248)))), 0xbeef);
     }
 
+    /// @dev Checks that the `loadAllocs` cheatcode persists account info if called inline
     function testLoadAllocsStatic() public {
+        // Restore the state snapshot prior to the allocs file being loaded.
+        vm.revertTo(snapshotId);
+
```

Comment:
@Evalir do we have confidence/good tests for snapshot/revertTo cheats? We don't verify that state was correctly reset after this call and I know there's been some bugs on those cheats in the cheat

Reply:
I did check it manually, but didn't commit them due to the tests in `Snapshots.t.sol`. Glad to add an explicit test for this in `loadAllocs.t.sol`, but figured we didn't want to cross-wires 😄

Reply:
Makes sense, I agree the tests for proper `revertTo` behavior are better suited elsewhere, so I'm ok leaving it as-is